client/src/components/Lesson/WordCreator/ContextSentences.tsx

```diff
@@ -46,37 +46,38 @@ export class ContextSentences extends React.Component<ContextSentencesProps, Con
             newVal = newVal.filter((sentence, i) =>
                 sentence !== '' || i === this.props.value.length);
             this.props.onChange(newVal);
-            if (ev.target) {
-                const result: number[] = [];
-                if (newVal[i] !== undefined) {
-                    result.push(newVal[i].search(this.props.wordInfo.word));
-                    this.props.wordInfo.forms.forEach((item: WordForm) => {result.push(newVal[i].search(item.word)); });
-                    if ( result.find((value: number) => value !== -1 ) !== undefined) {
-                        const error = Object.assign(this.state.error, {[i]: '',
-                        });
-                        this.props.onChangeError(error);
-                    }
-                    else {
-                        const error = Object.assign(this.state.error,
-                        {[i]: 'Sentence does not contain: '.concat(this.props.wordInfo.word).concat(' or one of its forms')
-                            });
-                        this.props.onChangeError(error);
-                    }
-                }
-                else {
-                    const error = Object.assign(this.state.error, {[i]: ''
-                        });
-                    this.props.onChangeError(error);
-                }
-            }
             return true;
         };
     }
-
+    validateSentence(sentence: string, i: number) {
+        const reg = new RegExp('\\b(' + this.props.wordInfo.word.toLocaleLowerCase() + '|' + this.props.wordInfo.forms.map(f => f.word.toLocaleLowerCase()).join('|') + ')\\b');
+        if (sentence !== undefined) {
+            if (reg.test(sentence.toLocaleLowerCase())) {
+                const error = Object.assign(this.state.error, {
+                [i]: '',
+                });
+                this.props.onChangeError(error);
+            }
+            else {
+                const error = Object.assign(this.state.error,
+                    {
+                    [i]: 'Sentence does not contain: '.concat(this.props.wordInfo.word).concat(' or one of its forms')
+                    });
+                this.props.onChangeError(error);
+            }
+        }
+        else {
```

Comment:
Can this `else` be combined with the `if` above that does the same thing? Maybe `if (sentence === undefined || reg.test(sentence.toLocaleLowerCase()))`?